### PR TITLE
support for call_indirect in the wasm parser

### PR DESCRIPTION
**Pull request by ColinEberhardt**
Follows a similar pattern to the way functions are parsed, the type information is looked up and added to the AST giving the same output as the WAT parser.

### CODE REVIEW

File: packages/wasm-edit/test/replace-node.js [hunk 1]
```diff
@@ -139,7 +139,7 @@ describe("replace a node", () => {
     const newBinary = edit(actualBinary, {
       Instr(path) {
         if (path.node.id === "get_global") {
-          const newNode = t.callIndirectInstructionIndex(t.indexLiteral(2));
+          const newNode = t.callIndirectInstructionIndex(t.indexLiteral(0));
```
Comment (ColinEberhardt) [inline packages/wasm-edit/test/replace-node.js]: I'm not sure `callIndirectInstructionIndex` should actually exist! I think our AST representation for call_indirect should always be populated with params / return - avoiding the need for different AST representations when parsing WASM vs. WAT

File: packages/wasm-parser/test/fixtures/func/anonymous-func-with-param/actual.wat-failure [hunk 2]
```diff
@@ -0,0 +1,7 @@
+;; named params have not been implemented in wasm parser yet
+(module 
+  (func (param $p i32))
+  (func (param $P i64))
+  (func (param $$ f32))
+  (func (param $$ f64))
+)
```
[the remark this answers is not in the document]
Comment (ColinEberhardt) [reply packages/wasm-parser/test/fixtures/func/anonymous-func-with-param/actual.wat-failure]: yes, these will be in the custom name section. So far I've only implemented the function sub-section:

https://webassembly.github.io/spec/core/appendix/custom.html#name-section

I plan to add the local name subsection shortly :-)

File: packages/wasm-parser/test/fixtures/func/with-shorthand-export/actual.wat-failure [hunk 3]
```diff
@@ -0,0 +1,6 @@
+;; shorthand export not implemented yet
```
[the remark this answers is not in the document]
Comment (ColinEberhardt) [reply packages/wasm-parser/test/fixtures/func/with-shorthand-export/actual.wat-failure]: Good point, that comment is incorrect. Both the WASM and WAT parser support this, but generate different ASTs. I'll make a change to ensure they are aligned.

File: packages/wasm-edit/test/replace-node.js [hunk 4]
```diff
@@ -157,7 +157,7 @@ describe("replace a node", () => {
       [constants.sections.type, 0x05, 0x01, 0x60, 0x00, 0x01, 0x7f],
       [constants.sections.func, 0x02, 0x01, 0x00],
       [constants.sections.global, 0x06, 0x01, 0x7f, 0x00, 0x41, 0x01, 0x0b],
-      [constants.sections.code, 0x07, 0x01, 0x05, 0x00, 0x11, 0x02, 0x00, 0x0b]
+      [constants.sections.code, 0x07, 0x01, 0x05, 0x00, 0x11, 0x00, 0x00, 0x0b]
```
Comment (xtuc) [inline packages/wasm-edit/test/replace-node.js]: It initially did this to avoid `0x00, 0x00` but it's fine to me.

**Pull request merged**